### PR TITLE
iOS: Canary release support + SPM automated releases

### LINKED ISSUE #407: iOS: SPM Canary Publishing
**Is your feature request related to a problem? Please describe.**

Canary builds generated on PRs should be consumable via SPM

**Describe the solution you'd like**

Add support for publishing canary versions of the swift package, since SPM is git reliant currently, this would mean a tag per canary release.

In order to keep things reasonable, the [release cleanup plugin](https://github.com/player-ui/player/blob/main/scripts/delete-old-prerelease.js) should be updated to also clean up old canary tags when their respective PRs have been released.

### PR DESCRIPTION
**Pull request by hborawski**
resolves #407 

### Change Type (required)
Indicate the type of change your pull request is:

- [x] `patch`
- [ ] `minor`
- [ ] `major`

### Does your PR have any documentation updates?
- [ ] Updated docs
- [ ] No Update needed
- [ ] Unable to update docs

<details>
  <summary>📦 Published PR as canary version: <code>0.7.5--canary.434.15236</code></summary>
  <br />

  Try this version out locally by upgrading relevant packages to 0.7.5--canary.434.15236
</details>

### CODE REVIEW

File: scripts/release.sh [hunk 1]
```diff
@@ -26,8 +26,7 @@ done
 # Rebuild to stamp the release podspec
 bazel build --config=release //:PlayerUI_Podspec //:PlayerUI_Pod
 
-# VScode extension publishing
-bazel run --config=release //language/vscode-player-syntax:vscode-plugin.publish
```
Comment (nancywu1) [inline scripts/release.sh:30]: should this VScode extension publishing be removed?

Comment (hborawski) [reply scripts/release.sh:30]: yeah this target is gone now, so it was just throwing errors

functionality was moved to the [tools repo](https://github.com/player-ui/tools/tree/main/language)

**Review by nancywu1 — APPROVED** [review]

**Pull request merged**